src/GameObject/EffectObject.ts

```diff
@@ -27,6 +28,7 @@ export default class EffectObject {
 	}
 
 	destroyOeder() {
-		this.effectObjectManager.destroy(this.effectObjectManager.effectObjects.indexOf(this));
+		// TODO Florian -- As we talked before, I'd rather pass `this`.
+		this.effectObjectManager.destroyEffectObject(this);
```

Comment:
加えて、こう書けばオブジェクト側がマネージャーを持つ必要もなくなるんですね。色々使ってて理解してきました。

Reply:
すみません、英語で説明させていただきます。

Yes it's better that you avoid accessing properties (members) from another object if you can, so that if you change the other object, you have less chances that everything breaks (example if you rename `EffectObjectManager.effectObjects` or make it another kind of list).

Reply:
ありがとうございます。理解できました。

今、これをもとに別のゲームを作っているのですが、
話を聞いてからコードを書き直すと、よく意味がわかりました。

あと、インターンで英語の重要性が身に染みたので、改めて勉強しています。
なので、（contextがわかる）英語を読む機会が増えてうれしいです。